### PR TITLE
Enhancement: Add hrefs to info widgets

### PR DESCRIPTION
**Pull request by DamitusThyYeetus123**
## Proposed change

Adds the `href: ...` property to info widgets (including logo, weather, etc.) to redirect on click.

Closes #2647 

## Type of change

- [ ] New service widget
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Documentation only
- [ ] Other (please explain)

## Checklist:

- [ ] If applicable, I have added corresponding documentation changes.
- [x] If applicable, I have reviewed the [feature](https://gethomepage.dev/more/development/#new-feature-guidelines) and / or [service widget guidelines](https://gethomepage.dev/more/development/#service-widget-guidelines).
- [x] I have checked that all code style checks pass using [pre-commit hooks](https://gethomepage.dev/more/development/#code-formatting-with-pre-commit-hooks) and [linting checks](https://gethomepage.dev/more/development/#code-linting).
- [x] If applicable, I have tested my code for new features & regressions on both mobile & desktop devices, using the latest version of major browsers.

### CODE REVIEW

**Review by shamoon — CHANGES_REQUESTED** [review]
Thanks, doesn't pass lint tho

File: src/components/widgets/datetime/datetime.jsx
Comment (shamoon) [inline src/components/widgets/datetime/datetime.jsx:1]: And isn't it redundant to have it here too?

Comment (DamitusThyYeetus123) [reply src/components/widgets/datetime/datetime.jsx:1]: Didn't realise that was also wrapped in container, will remove.

**Comment by shamoon** [comment]
Usually simplest to just make it an `<a>` tag

File: src/components/widgets/widget/container.jsx [hunk 1]
```diff
@@ -57,7 +57,10 @@ export function getBottomBlock(children) {
...
 export default function Container({ children = [], options, additionalClassNames = "" }) {
   return (
-    <div className={getAllClasses(options, `${additionalClassNames} widget-container`)}>
+    <div
+      onClick={() => (options.href ? window.location.assign(options.href) : [])}
+      className={getAllClasses(options, `${additionalClassNames} widget-container`)}
+    >
       {getInnerBlock(children)}
       {getBottomBlock(children)}
     </div>
```
Comment (shamoon) [inline src/components/widgets/widget/container.jsx:66]:
```suggestion
      options.href ? (
      <a href={options.href} className={getAllClasses(options, `${additionalClassNames} widget-container`)}>
        {getInnerBlock(children)}
        {getBottomBlock(children)}
      </a>
    ) : (
      <div className={getAllClasses(options, `${additionalClassNames} widget-container`)}>
        {getInnerBlock(children)}
        {getBottomBlock(children)}
      </div>
```

**Review by shamoon — CHANGES_REQUESTED** [review]

File: src/components/widgets/widget/container.jsx (hunk 1 above)
Comment (DamitusThyYeetus123) [reply src/components/widgets/widget/container.jsx:66]: yeah that looks good, I think I had some styling issues with `<a>`, but that might fix them.

**Review by shamoon — APPROVED** [review]

**Pull request merged**